### PR TITLE
Theme Showcase - Add trending tab to themes showcase

### PR DESCRIPTION
**Pull request by blackjackkent**
#### Changes proposed in this Pull Request

* This PR adds a "Trending" tab to the Themes Showcase. It follows the existing pattern for fetching recommended themes, but instead calls the v1.1 Themes endpoint, which supports a "sort" parameter which defaults to "trending". It also provides a small amount of general refactoring to future-proof this process for adding further tabs in the future.

#### Testing instructions

* Navigate to `/themes/{SITE_URL}` while logged in
* Verify that a "Trending" tab appears on the search tab bar.
* Verify that clicking on it presents a customized list of themes, different from that under "All" or "Recommended".

Related to #54088

### CODE REVIEW

File: client/my-sites/themes/theme-showcase.jsx [hunk 1]
```diff
@@ -59,6 +60,11 @@ const subjectsMeta = {
 	food: { icon: 'flip-horizontal', order: 9 },
 	music: { icon: 'audio', order: 10 },
 };
+const tabFilters = {
+	ALL: { key: 'all', text: 'All Themes' },
+	RECOMMENDED: { key: 'recommended', text: 'Recommended' },
+	TRENDING: { key: 'trending', text: 'Trending' },
+};
```
Comment (mreishus) [inline client/my-sites/themes/theme-showcase.jsx]: Text should be translated

File: client/my-sites/themes/theme-showcase.jsx [hunk 2]
```diff
@@ -373,6 +369,51 @@ class ThemeShowcase extends React.Component {
...
+									addTracking( options ),
+									( option ) => ! ( option.hideForTheme && option.hideForTheme( theme, siteId ) )
+								);
+							} }
+							trackScrollPage={ this.props.trackScrollPage }
+							emptyContent={ this.props.emptyContent }
+							scrollToSearchInput={ this.scrollToSearchInput }
+							bookmarkRef={ this.bookmarkRef }
+						/>
+					) }
```
Comment (mreishus) [inline client/my-sites/themes/theme-showcase.jsx:379]: Are these props mostly the same as the other ones? You may be able to reuse `{ ...themeProps }`. Check my initial creation here: https://github.com/Automattic/wp-calypso/pull/54629

File: client/my-sites/themes/themes-selection.jsx [hunk 3]
```diff
@@ -56,7 +56,7 @@ class ThemesSelection extends Component {
 		isRequesting: PropTypes.bool,
 		isThemeActive: PropTypes.func,
 		placeholderCount: PropTypes.number,
-		recommendedThemes: PropTypes.array,
+		customizedThemesList: PropTypes.array,
```
Comment (mreishus) [inline client/my-sites/themes/themes-selection.jsx:59]: :+1:

File: client/my-sites/themes/trending-themes.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+		}
+	}
+
+	fetchThemes() {
+		this.props.getTrendingThemes( this.props.filter );
+	}
+
+	render() {
+		return (
+			<>
```
Comment (mreishus) [inline client/my-sites/themes/trending-themes.jsx]: Nitpick: I don't believe this fragment is needed

File: client/my-sites/themes/theme-showcase.jsx (hunk 1 above)
Comment (blackjackkent) [reply client/my-sites/themes/theme-showcase.jsx]: Good catch, thank you!

File: client/my-sites/themes/theme-showcase.jsx (hunk 2 above)
Comment (blackjackkent) [reply client/my-sites/themes/theme-showcase.jsx:379]: Oh, sick! Yeah, that's a lot cleaner.

File: client/my-sites/themes/trending-themes.jsx (hunk 4 above)
Comment (blackjackkent) [reply client/my-sites/themes/trending-themes.jsx]: Good catch. I had c/p'd it from recommended-themes.jsx so I'll go ahead and remove it there too.

**Comment by blackjackkent** [comment]
Updated based on comments above.

File: client/my-sites/themes/theme-showcase.jsx [hunk 5]
```diff
@@ -191,7 +197,7 @@ class ThemeShowcase extends React.Component {
 	onTierSelect = ( { value: tier } ) => {
 		// In this state: tabFilter = [ ##Recommended## | All(1) ]   tier = [ All(2) | Free | Premium ]
 		// Clicking "Free" or "Premium" forces tabFilter from "Recommended" to "All"
-		if ( tier !== '' && tier !== 'all' && this.state.tabFilter === 'recommended' ) {
+		if ( tier !== '' && tier !== 'all' && this.state.tabFilter.key === 'recommended' ) {
 			this.setState( { tabFilter: 'all' } );
```
Comment (mreishus) [inline client/my-sites/themes/theme-showcase.jsx]: * Go to theme showcase
* Switch to recommended tab
* Click on the 'free' button on the right side
* Expected to see: all themes tab is selected. Actually see: No tab is selected

```suggestion
			this.setState( { tabFilter: this.tabFilters.ALL } );
```

File: client/my-sites/themes/theme-showcase.jsx [hunk 6]
```diff
@@ -207,7 +213,7 @@ class ThemeShowcase extends React.Component {
 		let callback = () => null;
 		// In this state: tabFilter = [ Recommended | ##All(1)## ]  tier = [ All(2) | Free | ##Premium## ]
 		// Clicking "Recommended" forces tier to be "all", since Recommend themes cannot filter on tier.
-		if ( 'recommended' === tabFilter && 'all' !== this.props.tier ) {
+		if ( 'recommended' === tabFilter.key && 'all' !== this.props.tier ) {
```
Comment (mreishus) [inline client/my-sites/themes/theme-showcase.jsx]: The logic here and on line 200 should be expanded from 'Is on recommended' to 'Is on recommended or on trending', or perhaps more generally, 'Is on any tab except for all themes'.

**Comment by mreishus** [comment]
Noted two above issues. Other than that, it works well, I only have one more question. When I switch to the trending tab, I see the blockbase and TT1 blocks themes. I haven't seen these display in the theme showcase before, so I'm wondering if they're not officially released yet? Will non-A8Cs see them as well and is this an issue?  @ianstewart might know more about this.

**Comment by ianstewart** [comment]
> Noted two above issues. Other than that, it works well, I only have one more question. When I switch to the trending tab, I see the blockbase and TT1 blocks themes. I haven't seen these display in the theme showcase before, so I'm wondering if they're not officially released yet? Will non-A8Cs see them as well and is this an issue? @ianstewart might know more about this.

This looks like an issue. Users shouldn't have access to them yet without being guided in via usability testing. @mattwiebe looks like we have another place where they'll show up.

**Comment by mreishus** [comment]
Well done, :+1:. This passes my review, with the one outstanding issue of yet non-released themes being available on the trending tab.

Two other notes:
* I did have some ideas about making the front-end only do one web request for all themes, then using client filtering logic to build the recommended, trending, and other tabs from that data. However, we're running against some timeline pressure, and I was having difficulty working with the data layer. Maybe we can return to this if we have some time later.
* I noticed that as the themes are loading, there is not a loading message, but rather it says 'no themes found'. However this also seems to be an existing issue on prod and can be improved in another step.

**Review by mreishus — APPROVED** [review]
:+1: LGTM

**Comment by blackjackkent** [comment]
> * I did have some ideas about making the front-end only do one web request for all themes, then using client filtering logic to build the recommended, trending, and other tabs from that data. However, we're running against some timeline pressure, and I was having difficulty working with the data layer. Maybe we can return to this if we have some time later.
> * I noticed that as the themes are loading, there is not a loading message, but rather it says 'no themes found'. However this also seems to be an existing issue on prod and can be improved in another step.

Agreed on both counts. This can definitely be further refined as more tabs get added.

**Comment by mreishus** [comment]
:tada:

**Pull request merged**